context/src/env/validator.rs

```diff
@@ -277,7 +277,7 @@ pub struct EnvValidationFlatIssue {
     pub error_message: String,
 }
 
-#[cfg_attr(feature = "pyo3", pymethods)]
+#[cfg_attr(feature = "pyo3", gen_stub_pymethods, pymethods)]
```

Comment:
a heads up to anyone reading this: order matters with `gen_stub_pymethods`—it needs to be before `pymethods` to generate types correctly. It's not documented and can cause some surprising results.